tools/c7n_index/c7n_index/metrics.py

```diff
@@ -61,23 +82,51 @@ class Indexer(object):
     """
 
 
-def get_indexer(config):
+def get_indexer(config, **kwargs):
     itype = config['indexer']['type']
     klass = indexers.get(itype)
-    return klass(config)
+    return klass(config, **kwargs)
+
+
+class MyConnection(RequestsHttpConnection):
+    def __init__(self, *args, **kwargs):
+        proxies = kwargs.pop('proxies', {})
+        super(MyConnection, self).__init__(*args, **kwargs)
+        self.session.proxies = proxies
 
 
 @indexers.register('es')
 class ElasticSearchIndexer(Indexer):
 
-    def __init__(self, config):
+    def __init__(self, config, **kwargs):
         self.config = config
-        self.client = Elasticsearch()
+        self.es_type = kwargs.get('type', 'policy-metric')
+
+        host = [config['indexer'].get('host', 'localhost')]
+        kwargs = {}
+
+        user = config['indexer'].get('user', False)
+        password = config['indexer'].get('password', False)
+        if user and password:
+            kwargs['http_auth'] = (user, password)
+
+        kwargs['port'] = config['indexer'].get('port', 9200)
+
+        proxy = config['indexer'].get('proxy', False)
```

Comment:
i'd rather leave this with proxies inherited from std environment variables, that removes the need for explicit config on them and the need for myconnection class.